packages/preferences/src/browser/views/components/single-preference-wrapper.tsx

```diff
@@ -133,24 +133,54 @@ export class SinglePreferenceWrapper extends React.Component<SinglePreferenceWra
     }
 
     protected renderOtherModifiedScopes(
+        id: string,
         preferenceValuesInAllScopes: Preference.ValuesInAllScopes | undefined,
-        data: PreferenceDataProperty,
-        currentScope: number): React.ReactNode[] | undefined {
+        currentScope: number,
+        service: PreferenceService): React.ReactNode[] | undefined {
         if (preferenceValuesInAllScopes) {
             return ['User', 'Workspace'].map((scope: 'User' | 'Workspace') => {
                 const matchingScope = PreferenceScope[scope];
-                const valueInCurrentScope = preferenceValuesInAllScopes[Preference.LookupKeys[currentScope] as keyof Preference.ValuesInAllScopes];
                 if (currentScope !== matchingScope) {
-                    const valueInOtherScope = preferenceValuesInAllScopes[Preference.LookupKeys[matchingScope] as keyof Preference.ValuesInAllScopes];
-                    if (valueInOtherScope && valueInOtherScope !== data.defaultValue) {
-                        const message = valueInCurrentScope && valueInCurrentScope !== data.defaultValue ? `Also modified in: ${scope}.` : `Modified in: ${scope}.`;
-                        return <i key={`modified-in-${scope}-alert`}><span> {message}</span></i>;
+                    const info = service.inspect<PreferenceItem>(id);
+                    if (!info) {
+                        return;
                     }
+
+                    const defaultValue = info.defaultValue;
+                    const currentValue = this.getValueByScope(info, currentScope);
+                    const matchingValue = this.getValueByScope(info, matchingScope);
+
+                    if ((currentValue !== matchingValue) && (typeof matchingValue !== 'undefined')) {
```

Comment:
When I run this code, if I change a preference to the same, non-default value in both User and Workspace scopes, I see no message, while VSCode shows a message in both scopes. I think we don't want to check that `currentValue !== matchingValue` before deciding to render something.

Reply:
@colin-grant-work yes you are right I had forgotten about that use-case.
I have adjusted the code and re-tested.